### PR TITLE
#nt 730

### PR DESCRIPTION
**Pull request by hasnik**
Added sorting capability to welcome-web page

### CODE REVIEW

File: src/pipes/sort-by-numeric-property.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { Pipe } from "@angular/core";
```
Comment (noro5) [inline src/pipes/sort-by-numeric-property.pipe.ts]: I think this pipe is not needed anymore and you can use the sortByProperty, which is an extended version. If so, then we should delete this pipe

Comment (noro5) [reply src/pipes/sort-by-numeric-property.pipe.ts]: I have noticed a strange behaviour when sorting byt Task:
image

File: src/pages/welcome-web/welcome-web.ts [hunk 2]
```diff
@@ -522,11 +525,60 @@ export class WelcomeWebPage {
...
+    return paes.sort(function (paeA, paeB) {
+        let paeADateEntry = paeA.getProjectActivityDateEntry(propertyToSortList);
+        let paeBDateEntry = paeB.getProjectActivityDateEntry(propertyToSortList);
+
+        let paeATimeQuantity = paeADateEntry.activityTimeEntryModels.length === 0 ? 0 : paeADateEntry.activityTimeEntryModels[0].quantity;
+        let paeBTimeQuantity = paeBDateEntry.activityTimeEntryModels.length === 0 ? 0 : paeBDateEntry.activityTimeEntryModels[0].quantity;
+
+        let quantityDifference = paeBTimeQuantity - paeATimeQuantity;
+
+        return reverseSortOrder
```
Comment (reg0) [inline src/pages/welcome-web/welcome-web.ts]: somehow I fell that it'd more readable like that:

```ts
return quantityDifference * (reverseSortOrder ? -1 : 1)
```

File: src/pipes/sort-by-property.pipe.ts [hunk 3]
```diff
@@ -2,19 +2,28 @@ import { Pipe } from "@angular/core";
...
 export class SortByPropertyPipe {
-    transform(array: any, arg: string): any {
-        if (array instanceof Array) {
-            array.sort((a: any, b: any) => {
-                if (a.hasOwnProperty(arg) && typeof a[arg] === "number" && b.hasOwnProperty(arg) && typeof b[arg] === "number") {
-                    return b[arg] - a[arg];
-                }
-                return 0;
-            });
+  transform(array: any, arg: string, reverseOrder: boolean): any {
```
Comment (reg0) [inline src/pipes/sort-by-property.pipe.ts]: reverseOrder could be an optional parameter

File: src/pipes/sort-by-property.pipe.ts [hunk 4]
```diff
@@ -2,19 +2,28 @@ import { Pipe } from "@angular/core";
...
+    if (array instanceof Array) {
+      array.sort((a: any, b: any) => {
+        if (a.hasOwnProperty(arg) && typeof a[arg] === "number" && b.hasOwnProperty(arg) && typeof b[arg] === "number") {
+          let difference = b[arg] - a[arg];
+
+          return reverseOrder
```
Comment (reg0) [inline src/pipes/sort-by-property.pipe.ts]: same as above: ```difference * (reverseOrder ? -1 : 1)```

File: src/pipes/sort-by-numeric-property.pipe.ts (hunk 1 above)
Comment (hasnik) [reply src/pipes/sort-by-numeric-property.pipe.ts]: Any steps to reproduce this order issue? I tried to do the same, but i'm unable to do that.

**Pull request merged**